### PR TITLE
chore: add contextual attributes to faro measurement

### PR DESCRIPTION
**Pull request by codecapitano**
#### PR Description
In the Faro Web-SDK we added an option to provide additional contextual attributes to the measurements payload.
To enable this we need to update the agent as well.

#### Notes to the Reviewer
Just for reference here are the PRs which make the needed changes to the Faro Receiver and the Faro Web-SDK.
* [PR Receiver](https://github.com/grafana/app-o11y-kwl-endpoint/pull/262)
* [PR Docs](https://github.com/grafana/website/pull/15059)

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Tests updated

### CODE REVIEW

File: pkg/integrations/v2/app_agent_receiver/payload.go [hunk 1]
```diff
@@ -207,11 +207,17 @@ func (l Log) KeyVal() *KeyVal {
...
+// represents the context of a log message
+type MeasurementContext map[string]string
+
 // Measurement holds the data for user provided measurements
 type Measurement struct {
 	Values    map[string]float64 `json:"values,omitempty"`
 	Timestamp time.Time          `json:"timestamp,omitempty"`
 	Trace     TraceContext       `json:"trace,omitempty"`
+	Context   MeasurementContext `json:"context,omitempty"`
+	Type      string             `json:"type,omitempty"`
```
Comment (rlankfo) [inline pkg/integrations/v2/app_agent_receiver/payload.go:220]: `Type` doesn't appear to be included in the `KeyVal` object. Do we want to add this to `func (m Measurement) KeyVal() *KeyVal` ? 

```go
KeyValAdd(kv, "type", m.Type)
```

**Review by rlankfo — APPROVED** [review]
LGTM for the `MeasurementContext`.  I had one question about adding `Type` but I'm not sure if it's necessary in this PR.

**Comment by rlankfo** [comment]
@codecapitano looks like there are some additional test cases we should update:

```
--- FAIL: TestExportLogs (0.01s)
    logs_exporter_test.go:103: 
        	Error Trace:	/Users/runner/work/agent/agent/pkg/integrations/v2/app_agent_receiver/logs_exporter_test.go:103
        	Error:      	Not equal: 
        	            	expected: "timestamp=\"2021-09-30 10:46:17.68 +0000 UTC\" kind=measurement ttfb=14.000000 ttfcp=22.120000 ttfp=20.120000 traceID=abcd spanID=def sdk_name=grafana-frontend-agent sdk_version=1.0.0 app_name=testapp app_release=0.8.2 app_version=abcdefg app_environment=production user_email=geralt@kaermorhen.org user_id=123 user_username=domasx2 user_attr_foo=bar session_id=abcd session_attr_time_elapsed=100s page_url=https://example.com/page browser_name=chrome browser_version=88.12.1 browser_os=linux browser_mobile=false view_name=foobar"
        	            	actual  : "timestamp=\"2021-09-30 10:46:17.68 +0000 UTC\" kind=measurement ttfb=14.000000 ttfcp=22.120000 ttfp=20.120000 traceID=abcd spanID=def context_hello=world sdk_name=grafana-frontend-agent sdk_version=1.0.0 app_name=testapp app_release=0.8.2 app_version=abcdefg app_environment=production user_email=geralt@kaermorhen.org user_id=123 user_username=domasx2 user_attr_foo=bar session_id=abcd session_attr_time_elapsed=100s page_url=https://example.com/page browser_name=chrome browser_version=88.12.1 browser_os=linux browser_mobile=false view_name=foobar"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-timestamp="2021-09-30 10:46:17.68 +0000 UTC" kind=measurement ttfb=14.000000 ttfcp=22.120000 ttfp=20.120000 traceID=abcd spanID=def sdk_name=grafana-frontend-agent sdk_version=1.0.0 app_name=testapp app_release=0.8.2 app_version=abcdefg app_environment=production user_email=geralt@kaermorhen.org user_id=123 user_username=domasx2 user_attr_foo=bar session_id=abcd session_attr_time_elapsed=100s page_url=https://example.com/page browser_name=chrome browser_version=88.12.1 browser_os=linux browser_mobile=false view_name=foobar
        	            	+timestamp="2021-09-30 10:46:17.68 +0000 UTC" kind=measurement ttfb=14.000000 ttfcp=22.120000 ttfp=20.120000 traceID=abcd spanID=def context_hello=world sdk_name=grafana-frontend-agent sdk_version=1.0.0 app_name=testapp app_release=0.8.2 app_version=abcdefg app_environment=production user_email=geralt@kaermorhen.org user_id=123 user_username=domasx2 user_attr_foo=bar session_id=abcd session_attr_time_elapsed=100s page_url=https://example.com/page browser_name=chrome browser_version=88.12.1 browser_os=linux browser_mobile=false view_name=foobar
        	Test:       	TestExportLogs
```

**Review by rlankfo — CHANGES_REQUESTED** [review]
test cases + changelog formatting

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,22 +7,21 @@ This document contains a historical list of changes between releases. Only
 changes that impact end-user behavior are listed; changes to documentation or
 internal API changes are not present.
 
-Main (unreleased)
------------------
+## Main (unreleased)
```
Comment (rlankfo) [inline CHANGELOG.md:10]: looks like the changelog was formatted. prettier on save? :D  we should make sure we only add the entry we need to this file.

File: pkg/integrations/v2/app_agent_receiver/payload.go [hunk 3]
```diff
@@ -207,11 +207,17 @@ func (l Log) KeyVal() *KeyVal {
 	return kv
 }
 
+// ExceptionContext is a string to string map structure that
```
Comment (marctc) [inline pkg/integrations/v2/app_agent_receiver/payload.go:210]: This comment doesn't match the name of the type

**Review by rlankfo — APPROVED** [review]
lgtm!

**Pull request merged**